src/planner.c

```diff
@@ -739,11 +741,12 @@ timescale_create_upper_paths_hook(PlannerInfo *root, UpperRelationKind stage, Re
 		/* Modify for INSERTs on a hypertable */
 		if (output_rel->pathlist != NIL)
 			output_rel->pathlist = replace_hypertable_insert_paths(root, output_rel->pathlist);
-
-		if (parse->hasAggs && stage == UPPERREL_GROUP_AGG)
+		if (stage == UPPERREL_GROUP_AGG && parse->hasAggs)
```

Comment:
Why the reordering? Can `parse` be null when `stage` is not `UPPERREL_GROUP_AGG`?

Reply:
reverted. left over from some code changes that I tried out.